### PR TITLE
Core: MIPS: Complete the support of mips64el and mips64r2el

### PR DESCRIPTION
**Pull request by FlyGoat**
MIPS changes.

### CODE REVIEW

**Comment by liushuyu** [comment]
Do these patches have negative effect on other architectures?

**Comment by MingcongBai** [comment]
> Do these patches have negative effect on other architectures?

I wouldn't believe so since they are all limited to the domains of MIPS architectures.

**Comment by liushuyu** [comment]
Requested review from @lmy441900

File: core-libs/glibc/autobuild/prepare [hunk 1]
```diff
@@ -5,7 +5,14 @@ git clone --depth 1 -b v"$LOCALEGENVER" \
     https://github.com/AOSC-Dev/locale-gen
 
 # MIPS64el specific.
-if [[ "${CROSS:-$ARCH}" = "mips64el" ]]; then
-    export CFLAGS="${CFLAGS} -march=mips64r2 -mabi=64 -mrelax-pic-calls"
-    export CXXFLAGS="${CXXFLAGS} -march=mips64r2 -mabi=64 -mrelax-pic-calls"
+if [[ "${CROSS:-$ARCH}" = "mips64" ]]; then
```
Comment (jyhi) [inline core-libs/glibc/autobuild/prepare]: Wait, why not `mips64el` here.

**Review by jyhi — CHANGES_REQUESTED** [review]

File: core-devel/binutils/autobuild/patches/0005-bfd-pic.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Index: b/bfd/Makefile.am
+===================================================================
+--- a/bfd/Makefile.am
++++ b/bfd/Makefile.am
+@@ -51,7 +51,7 @@ ZLIBINC = @zlibinc@
+ 
+ WARN_CFLAGS = @WARN_CFLAGS@
+ NO_WERROR = @NO_WERROR@
+-AM_CFLAGS = $(WARN_CFLAGS) $(ZLIBINC)
++AM_CFLAGS = $(WARN_CFLAGS) $(ZLIBINC) -fPIC
```
Comment (FlyGoat) [inline core-devel/binutils/autobuild/patches/0005-bfd-pic.patch:13]: Well, I'm concerning this patch may have effect on other architecture. It's grabbed from Debian and probably necessary for MIPS64.

Comment (MingcongBai) [reply core-devel/binutils/autobuild/patches/0005-bfd-pic.patch:13]: Then you should consider adding a `.mips64el` suffix or something that matches the architecture name - making it `0005-bfd-pic.patch.mips64el`.

**Comment by liushuyu** [comment]
Don't know if this is still valid. If so, someone still needs to clean up this mess and rebase the changes against the current HEAD.

**Comment by jyhi** [comment]
@liushuyu If @FlyGoat does not do the cleanups then we may restart the MIPS port in a few months.

**Comment by FlyGoat** [comment]
@liushuyu Will do later. I've just done the toolchain porting and now it time to build core again...

**Comment by jyhi** [comment]
Okay, in the upcoming Core 6 we will restart MIPS port with a different configuration, so this PR should be closed as @FlyGoat does not respond.

**Pull request closed without merging**